### ISSUE TITLE
[DOC] "build requirements" section in installation section, may be missing content or may be better moved

### ISSUE THREAD

**Issue by fkiraly** [alan-turing-institute/sktime#2329]
There is an inconsistency in the developer install instructions:
at the top, it is said that step 2, the "build requirements" section, depends on the operating system.

However, there is no distinction between different operating systems in the "build requirements" section, and it also seems to be specific to windows?
It is also substantially shortened, I believe since we removed the dependency on C, so some content might have been cut erroneously, or it might have been cut and now step 2 is much simpler.

In either case, we need to review this:
* is step 2 no longer operating system dependent? If so, how do we deal with windows specific stuff such as WSL?
* possibly move step 2 in sequence with step 1 and step 3 since step 2 is no longer highly complicated (since removal of C)

FYI @lmmentel, did you make the change that shortens step 2?

**Comment by lmmentel** [comment]
@fkiraly could you paste a link to the docs or mention the file name this is related to?

**Comment by fkiraly** [comment]
oh, of course, silly me. Added the link in the issue description.

**Comment by lmmentel** [comment]
Yeah I remember cleaning a lot of that up after C extensions were dropped. We do not really have any system deps as long as you have a compatible python version. That being said some of our dependencies have system requirements especially on exotic platforms, e.g. a lot of C dependent packages need to be compiled on apple silicon since there are no wheels yet. I would delegate the responsibility to provide accurate installation instructions to those packages.

**Comment by fkiraly** [comment]
well, then we need to clean up the sections that reference possible branching paths, and probably move the sections in order, since section 2 is linear and small now?

**Comment by lmmentel** [comment]
?
